app/assets/javascripts/controllers/CommonChatController.js

```diff
@@ -380,21 +379,32 @@ export default class CommonChatController {
         this.sdk.sendActivityMessage("stopTyping");
     }
 
+    hasBeenSurveyed() {
+    	return document.cookie.includes("surveyed=true")
+	}
+
     onConfirmEndChat() {
         this.closeNuanceChat();
         let escalated = this.state.isEscalated();
 
-        this._moveToClosingState();
+        console.log("ending chat")
 
-        if (escalated) {
-            this._sendPostChatSurveyWebchat(this.sdk).beginPostChatSurvey(webchatSurvey, automatonWebchat, timestamp);
-            this.container.showPage(new PostChatSurveyWebchat((page) => this.onPostChatSurveyWebchatSubmitted(page)));
-        } else {
-            this._sendPostChatSurveyDigitalAssistant(this.sdk).beginPostChatSurvey(digitalAssistantSurvey, automatonDA, timestamp);
-            this.container.showPage(new PostChatSurveyDigitalAssistant((page) => this.onPostChatSurveyDigitalAssistantSubmitted(page)));
-        }
+		this._moveToClosingState();
+
+		if (this.hasBeenSurveyed()) {
+			this.showEndChatPage(false);
+		} else {
+			document.cookie = "surveyed=true";
+			if (escalated) {
+				this._sendPostChatSurveyWebchat(this.sdk).beginPostChatSurvey(webchatSurvey, automatonWebchat, timestamp);
+				this.container.showPage(new PostChatSurveyWebchat((page) => this.onPostChatSurveyWebchatSubmitted(page)));
+			} else {
+				this._sendPostChatSurveyDigitalAssistant(this.sdk).beginPostChatSurvey(digitalAssistantSurvey, automatonDA, timestamp);
+				this.container.showPage(new PostChatSurveyDigitalAssistant((page) => this.onPostChatSurveyDigitalAssistantSubmitted(page)));
+			}
+			window.GOVUKFrontend.initAll();
```

Comment:
perhaps a little comment about what this line does

Reply:
do we still need the console.log?